rpc/v2_endpoints.go

```diff
@@ -380,3 +381,178 @@ func (s *Server) GetIdentityTransfersInTickRangeV2(ctx context.Context, req *pro
+				QuorumTickStructure:   tickData.QuorumTickStructure,
+				QuorumDiffPerComputor: make(map[uint32]*protobuff.QuorumDiff),
+			},
+		}
+
+		for id, diff := range tickData.QuorumDiffPerComputor {
+			res.QuorumTickData.QuorumDiffPerComputor[id] = &protobuff.QuorumDiff{
+				ExpectedNextTickTxDigestHex: diff.ExpectedNextTickTxDigestHex,
+				SignatureHex:                diff.SignatureHex,
+			}
+		}
+
+		return &res, nil
+	}
+
+	nextTick := req.TickNumber + 1
+
+	//Check if next tick was skipped
+	//TODO: check if this is accurate behaviour (if the tick gets skipped, then the data should be in the next available tick)
+	skipped, nextAvailable := wasTickSkippedByArchive(nextTick, processedTickIntervalsPerEpoch)
+	if skipped {
+		nextTick = nextAvailable
+	}
+
+	//Get quorum data for next tick
+	nextTickQuorumData, err := s.store.GetQuorumTickDataV2(ctx, nextTick)
+	if err != nil {
+		if errors.Is(err, store.ErrNotFound) {
+			return nil, status.Errorf(codes.Internal, "quorum data for next tick was not found")
+		}
+		return nil, status.Errorf(codes.Internal, "getting tick data: %v", err)
+	}
+
+	//Get quorum data for current tick
+	currentTickQuorumData, err := s.store.GetQuorumTickDataV2(ctx, req.TickNumber)
+	if err != nil {
+		if errors.Is(err, store.ErrNotFound) {
+			return nil, status.Errorf(codes.Internal, "quorum data for  tick was not found")
+		}
+		return nil, status.Errorf(codes.Internal, "getting tick data: %v", err)
+	}
+
+	//Get computors
+	computors, err := s.store.GetComputors(ctx, currentTickQuorumData.QuorumTickStructure.Epoch)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "getting computor list")
+	}
+
+	//Response object
+	res := protobuff.GetQuorumTickDataResponse{
+		QuorumTickData: &protobuff.QuorumTickData{
+			QuorumTickStructure:   currentTickQuorumData.QuorumTickStructure,
+			QuorumDiffPerComputor: make(map[uint32]*protobuff.QuorumDiff),
+		},
+	}
+
+	//Digests
+	spectrumDigest, err := hex.DecodeString(nextTickQuorumData.QuorumTickStructure.PrevSpectrumDigestHex)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "obtaining spectrum digest from next tick quorum data")
+	}
+	universeDigest, err := hex.DecodeString(nextTickQuorumData.QuorumTickStructure.PrevUniverseDigestHex)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "obtaining universe digest from next tick quorum data")
+	}
+	computerDigest, err := hex.DecodeString(nextTickQuorumData.QuorumTickStructure.PrevComputerDigestHex)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "obtaining computer digest from next tick quorum data")
+	}
+	resourceDigest, err := hex.DecodeString(nextTickQuorumData.QuorumTickStructure.PrevResourceTestingDigestHex)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "obtaining resource testing digest from next tick quorum data")
+	}
+
+	//Loop over all computors in current tick data
+	for id, voteDiff := range currentTickQuorumData.QuorumDiffPerComputor {
```

Comment:
I believe this entire reconstruction logic should reside in a separate package; maybe validator